openfl/cryptography/io.py

```diff
@@ -111,3 +111,21 @@ def read_csr(path: Path) -> Tuple[CertificateSigningRequest, str]:
     # TODO: replace assert with exception / sys.exit
     assert (isinstance(csr, x509.CertificateSigningRequest))
     return csr, hasher.hexdigest()
+
+
+def get_csr_hash(certificate: CertificateSigningRequest) -> str:
```

Comment:
I like the separation of the CSR hashing functionality. I would just recommend refactoring the `read_csr` function to reuse the `get_csr_hash` function internally